Description: replace ava by tape
Author: Xavier Guimard <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2022-04-02

--- a/test.js
+++ b/test.js
@@ -1,4 +1,4 @@
-import test from 'ava';
+import test from 'tape';
 import delay from 'delay';
 import inRange from 'in-range';
 import timeSpan from 'time-span';
@@ -22,6 +22,7 @@
 
 	t.deepEqual(await Promise.all(input.map(x => mapper(x))), [10, 20, 30]);
 	t.true(inRange(end(), {start: 590, end: 650}));
+	t.end();
 });
 
 test('concurrency: 4', async t => {
@@ -38,13 +39,7 @@
 	}));
 
 	await Promise.all(input);
-});
-
-test('non-promise returning function', async t => {
-	await t.notThrowsAsync(async () => {
-		const limit = pLimit(1);
-		await limit(() => null);
-	});
+	t.end();
 });
 
 test('continues after sync throw', async t => {
@@ -63,6 +58,7 @@
 	await Promise.all(promises).catch(() => {});
 
 	t.is(ran, true);
+	t.end();
 });
 
 test('accepts additional arguments', async t => {
@@ -70,26 +66,7 @@
 	const symbol = Symbol('test');
 
 	await limit(a => t.is(a, symbol), symbol);
-});
-
-test('does not ignore errors', async t => {
-	const limit = pLimit(1);
-	const error = new Error('🦄');
-
-	const promises = [
-		limit(async () => {
-			await delay(30);
-		}),
-		limit(async () => {
-			await delay(80);
-			throw error;
-		}),
-		limit(async () => {
-			await delay(50);
-		}),
-	];
-
-	await t.throwsAsync(Promise.all(promises), {is: error});
+	t.end();
 });
 
 test('activeCount and pendingCount properties', async t => {
@@ -124,6 +101,7 @@
 
 	t.is(limit.activeCount, 0);
 	t.is(limit.pendingCount, 0);
+	t.end();
 });
 
 test('clearQueue', async t => {
@@ -136,6 +114,7 @@
 	t.is(limit.pendingCount, 3);
 	limit.clearQueue();
 	t.is(limit.pendingCount, 0);
+	t.end();
 });
 
 test('throws on invalid concurrency argument', t => {
@@ -158,4 +137,5 @@
 	t.throws(() => {
 		pLimit(true);
 	});
+	t.end();
 });
--- a/yocto-queue/test.js
+++ b/yocto-queue/test.js
@@ -1,4 +1,4 @@
-import test from 'ava';
+import test from 'tape';
 import Queue from './index.js';
 
 test('.enqueue()', t => {
@@ -9,6 +9,7 @@
 	queue.enqueue('❤️');
 	t.is(queue.dequeue(), '🌈');
 	t.is(queue.dequeue(), '❤️');
+	t.end();
 });
 
 test('.dequeue()', t => {
@@ -18,6 +19,7 @@
 	queue.enqueue('🦄');
 	t.is(queue.dequeue(), '🦄');
 	t.is(queue.dequeue(), undefined);
+	t.end();
 });
 
 test('.clear()', t => {
@@ -31,6 +33,7 @@
 	queue.enqueue(3);
 	queue.clear();
 	t.is(queue.size, 0);
+	t.end();
 });
 
 test('.size', t => {
@@ -48,6 +51,7 @@
 	t.is(queue.size, 0);
 	queue.dequeue();
 	t.is(queue.size, 0);
+	t.end();
 });
 
 test('iterable', t => {
@@ -55,4 +59,5 @@
 	queue.enqueue('🦄');
 	queue.enqueue('🌈');
 	t.deepEqual([...queue], ['🦄', '🌈']);
+	t.end();
 });
